### ISSUE TITLE
Inconsistency in border shorthands

### ISSUE THREAD

**Issue by QingqiShi** [facebook/stylex#127]
**The problem**

I've encountered an issue with the use of border shorthands in the Stylex package. While the ESLint rule provided by Stylex seems to prohibit the use of shorthand properties for borders, there's an inconsistency in how these rules are applied and enforced in practice.

I used border shorthand properties (`borderTop`, `borderLeft`, `borderRight`, `borderBottom`) in my Stylex code as follows:

```js
const s = stylex.create({
  test: {
    borderTop: '5px solid blue',
    borderLeft: '5px solid blue',
    borderRight: '5px solid blue',
    borderBottom: '5px solid blue',
  },
});
```

When running `npm run lint`, the following errors are reported:

```
./app/page.tsx
9:5  Error: The 'borderTop' property is not supported. Use the 'borderTopWidth', 'borderTopStyle' and 'borderTopColor' properties instead.  @stylexjs/valid-styles
10:5  Error: The 'borderLeft' property is not supported. Use the 'borderLeftWidth', 'borderLeftStyle' and 'borderLeftColor' properties instead.  @stylexjs/valid-styles
11:5  Error: The 'borderRight' property is not supported. Use the 'borderRightWidth', 'borderRightStyle' and 'borderRightColor' properties instead.  @stylexjs/valid-styles
12:5  Error: The 'borderBottom' property is not supported. Use the 'borderBottomWidth', 'borderBottomStyle' and 'borderBottomColor' properties instead.  @stylexjs/valid-styles
```

However, in the rendered output, `borderTop`, `borderRight`, and `borderBottom` are actually applied and rendered correctly, but `borderLeft` is not.

<img width="298" alt="截屏2023-12-11 下午7 23 03" src="https://github.com/facebook/stylex/assets/6232457/31c3ceda-3ea8-404b-914d-7c3a815db932">

**How to reproduce**

I have created a StackBlitz reproduction of this issue, which can be found here: https://stackblitz.com/edit/stylex-next-4kubjy?file=app%2Fpage.tsx

**Expected behavior**

I would expect the ESLint rule to either allow all border shorthand properties or to consistently enforce the prohibition across all of them. The current behavior where some properties work despite the ESLint errors, and others do not, is confusing.

Could you please provide some clarification on this behavior? Is this an intended feature or an oversight in the ESLint rule implementation?

Thank you for your assistance.

**Environment (include versions). Did this work in previous versions?**

Mac and Linux, v0.3.0 of stylex, latest Next.js.

**Comment by nmn** [comment]
> but borderLeft is not

This is indeed strange. 

- All border shorthands should apply.
- All of them should give you a lint error. (Reconsider this decision)

**Comment by QingqiShi** [comment]
@nmn, thank you for the prompt response! I really appreciate it. Additionally, I believe it would be beneficial if the documentation could clarify why these shorthand properties trigger a lint error, especially if they are intended to be supported in practice. This would greatly aid developers in understanding the intended usage and limitations.

**Comment by nmn** [comment]
@QingqiShi We initially disallowed a few shorthand properties:
- `background`
- `animation`
- `transition`
- `border` and `border[Dir]`

This was done for two reasons:
1. Alignment with React Native
2. A conceptual idea that one property should have one value. (We discourage multiple values for `margin` for example)

We're reconsidering this decision and will document based on where we land.

**Comment by nmn** [comment]
This particular issue has been resolved and a separate issue is tracking other missing properties. So closing.
